pandas/_libs/reshape.pyx

```diff
@@ -80,7 +80,7 @@ def unstack(numeric_object_t[:, :] values, const uint8_t[:] mask,
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def explode(ndarray[object] values):
```

Comment:
can we get here with a readonly ndarray?

Reply:
Looks like in our testing this array is always writable (from `mgr.internal_values()`), should this be readonly?

Reply:
I think it can depend on what user passed to the constructor

Reply:
Gotcha. Well specifically here, we can't use `const` as that's not supported with object memoryviews